### PR TITLE
chore: update config

### LINKED ISSUE #790: Move shared constants to default config file.
@0xcodercrane shouldn't these all be moved to `ubiquibot-config-default.ts` ? If so, please file a new issue or just handle it. Thanks!

_Originally posted by @pavlovcik in https://github.com/ubiquity/ubiquibot/pull/748#discussion_r1332052611_

### PR DESCRIPTION
**Pull request by molecula451**
Resolves #790 

Quality Assurance:

### CODE REVIEW

**Review by 0x4007 — DISMISSED** [review]
Looks pretty good to me although I don't deeply understand all this code's context.

**Comment by whilefoo** [comment]
@pavlovcik this PR puts all those shared constants in config which means they can be configured by `ubuiquibot-config.yaml`. Do we really want that?

**Comment by molecula451** [comment]
> @pavlovcik this PR puts all those shared constants in config which means they can be configured by `ubuiquibot-config.yaml`. Do we really want that?

pavlocik asked, https://github.com/ubiquity/ubiquibot/pull/748#discussion_r1332052611

**Comment by pavlovcik** [comment]
> @pavlovcik this PR puts all those shared constants in config which means they can be configured by `ubuiquibot-config.yaml`. Do we really want that?

Why is it a bad idea to let our partners configure these things? 
These only affect their repositories, right?

**Comment by whilefoo** [comment]
> Why is it a bad idea to let our partners configure these things?
> These only affect their repositories, right?

Permit base url and bot delay probably shouldn't be modified, other things are okay I guess.

File: src/utils/private.ts [hunk 1]
```diff
@@ -161,6 +161,13 @@ export const getWideConfig = async (context: Context) => {
     openAIKey: mergedConfigData.openAIKey,
     openAITokenLimit: mergedConfigData.openAITokenLimit,
     staleBountyTime: mergedConfigData.staleBountyTime,
+    startCommandEnabled: mergedConfigData.startCommandEnabled,
```
Comment (whilefoo) [inline src/utils/private.ts:164]: Why do we have a separate `startCommandEnabled` if we already have `commandSettings`?

File: src/utils/private.ts [hunk 2]
```diff
@@ -161,6 +161,13 @@ export const getWideConfig = async (context: Context) => {
...
+    defaultTimeRangeForMaxIssue: mergedConfigData.defaultTimeRangeForMaxIssue,
```
Comment (whilefoo) [inline src/utils/private.ts:165]: timeRangeForMaxIssue is a confusing name, I can't really tell what it means just from the name.

File: src/utils/private.ts [hunk 3]
```diff
@@ -161,6 +161,13 @@ export const getWideConfig = async (context: Context) => {
...
+    defaultTimeRangeForMaxIssueEnabled: mergedConfigData.defaultTimeRangeForMaxIssueEnabled,
+    defaultPermitBaseUrl: mergedConfigData.defaultPermitBaseUrl,
+    defaultBotDelay: mergedConfigData.defaultBotDelay,
+    defaultFollowUpTime: mergedConfigData.defaultFollowUpTime,
+    defaultDisqualifyTime: mergedConfigData.defaultDisqualifyTime,
```
Comment (whilefoo) [inline src/utils/private.ts:170]:
```suggestion
    disqualifyTime: mergedConfigData.disqualifyTime,
```
they shouldn't have `default` in front of them

**Review by whilefoo — CHANGES_REQUESTED** [review]

**Comment by molecula451** [comment]
> > Why is it a bad idea to let our partners configure these things?
> > These only affect their repositories, right?
> 
> Permit base url and bot delay probably shouldn't be modified, other things are okay I guess. 
> 

But they could use the defacto value. The thing is getting rid of `shared.ts` no?

File: src/utils/private.ts (hunk 1 above)
Comment (molecula451) [reply src/utils/private.ts:164]: Can You provide more context?

`startCommandEnabled` is for /start as it makes more sense than `assignCommandEnabled`

Comment (molecula451) [reply src/utils/private.ts:164]: > Can You provide more context?
> 
> `startCommandEnabled` is for `/start` as it makes more sense than `assignCommandEnabled`

Comment (molecula451) [reply src/utils/private.ts:164]: Can You provide more context?

 `startCommandEnabled` is for `/start` as it makes more sense than `assignCommandEnabled`

File: src/utils/private.ts (hunk 2 above)
Comment (molecula451) [reply src/utils/private.ts:165]: This is the same name as original in `shared.ts` suggestions are welcome if wey want it Changed it

File: src/utils/private.ts (hunk 1 above)
Comment (whilefoo) [reply src/utils/private.ts:164]: `commandSettings` contains the `/start` command and you can enable or disable it there.

**Comment by pavlovcik** [comment]
> Permit base url and bot delay

Well the base url is just an RPC endpoint. I dont see the issue with partners setting their own dedicated RPC endpoints if they really need it for some reason. 

What is bot delay for?

File: src/utils/private.ts (hunk 2 above)
Comment (pavlovcik) [reply src/utils/private.ts:165]: > timeRangeForMaxIssue is a confusing name, I can't really tell what it means just from the name.

What does it do? Let's rename it to something more expressive.

**Comment by whilefoo** [comment]
> Well the base url is just an RPC endpoint. I dont see the issue with partners setting their own dedicated RPC endpoints if they really need it for some reason.

It's not RPC endpoint, it is the claim url `https://pay.ubq.fi`

> What is bot delay for?

it seems it's for the telegram feature, as you said we can remove this feature but if someone sets it to a large number then the bot will timeout.

**Comment by pavlovcik** [comment]
> > Well the base url is just an RPC endpoint. I dont see the issue with partners setting their own dedicated RPC endpoints if they really need it for some reason.
> 
> It's not RPC endpoint, it is the claim url `https://pay.ubq.fi`
> 

Whitelabeling the claims portal seems pretty fancy but perhaps not in our best interests. It can be a secret config setting lol


> > What is bot delay for?
> 
> it seems it's for the telegram feature, as you said we can remove this feature but if someone sets it to a large number then the bot will timeout.

We should completely wipe that stuff then it all seems like problematic code.

File: src/utils/private.ts (hunk 2 above)
Comment (whilefoo) [reply src/utils/private.ts:165]: It's used when checking if a bounty hunter can assign a new issue and it checks how many issues is already assigned to them and how many pull requests are approved or with no reviews but X time old. `TimeRangeForMaxIssue` is X.

Comment (0x4007) [reply src/utils/private.ts]: What should we rename it to?

**Comment by molecula451** [comment]
>  What is bot delay for? 
>  it seems it's for the telegram feature, as you said we can remove this feature but if someone sets it to a large number then the bot will timeout.

> We should completely wipe that stuff then it all seems like problematic code.

I have gotten rid of telegram stuff code on config at https://github.com/ubiquity/ubiquibot/pull/802/commits/6254917721ec79987dd6c8605b056c2c466ccd90 review accordingly

**Comment by molecula451** [comment]
> We should completely wipe that stuff then it all seems like problematic code.

The telegram stuff seems out of scope for this PR, needs greater refactor

File: src/utils/private.ts (hunk 2 above)
Comment (molecula451) [reply src/utils/private.ts]: this where the gpt /ask would come in handy for all to see

/ask can you provide some names for timeRangeForMaxIssue var

Comment (molecula451) [reply src/utils/private.ts]: > timeRangeForMaxIssue is a confusing name, I can't really tell what it means just from the name.


Certainly! Here is a list of similar names for the variable "timeRangeForMaxIssue":

maxIssueTimeRange
issueTimeRangeMax
maxIssueDuration
timeFrameForMaxProblem
peakIssueTimeSpan
maximumIssueTimeFrame
issuePeakWindow
peakProblemTimeRange
maxIssuePeriod
timeSpanForPeakIssues

**Comment by molecula451** [comment]
`shared.ts` it's purged

**Comment by molecula451** [comment]
@wannacfuture might want to help pavlovcik debug and Review this simple PR?

**Comment by wannacfuture** [comment]
Would you mind resolving the conflicts?

Let's get this merged before more config conflicts occur. @pavlovcik 

@molecula451 , Please post a QA.

**Comment by molecula451** [comment]
I do not see an issue with conflicts. The only conflicts shows it's shared.ts but shared.ts it getting purged in this PR there shouldn't be issue

**Comment by molecula451** [comment]
> I do not see an issue with conflicts. The only conflicts shows it's `shared.ts` but `shared.ts` it getting purged in this PR there shouldn't be issue

**Comment by molecula451** [comment]
I do not see an issue with conflicts. The only conflicts shows it's `shared.ts` but `shared.ts` it getting purged in this PR there shouldn't be issue

**Pull request closed without merging**